digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataexport/ExportDataApi.scala

```diff
@@ -0,0 +1,101 @@
+package fi.liikennevirasto.digiroad2.dataexport
+
+import fi.liikennevirasto.digiroad2.asset.DateParser
+import fi.liikennevirasto.digiroad2.asset.DateParser.DateTimePropertyFormat
+import fi.liikennevirasto.digiroad2.authentication.RequestHeaderAuthentication
+import fi.liikennevirasto.digiroad2.csvDataExporter.AssetReportCsvExporter
+import fi.liikennevirasto.digiroad2.dao.ExportReportDAO
+import fi.liikennevirasto.digiroad2.middleware.CsvDataExporterInfo
+import fi.liikennevirasto.digiroad2.service.RoadLinkService
+import fi.liikennevirasto.digiroad2.{CsvDataExporter, Digiroad2Context, DigiroadEventBus}
+import fi.liikennevirasto.digiroad2.user.UserProvider
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+import org.json4s.{CustomSerializer, DefaultFormats, Formats, JString}
+import org.scalatra.{Forbidden, ScalatraServlet, Unauthorized}
+import org.scalatra.json.JacksonJsonSupport
+
+
+class ExportDataApi( roadLinkService: RoadLinkService, userProvider: UserProvider = Digiroad2Context.userProvider, val eventBus: DigiroadEventBus = Digiroad2Context.eventbus)
+extends ScalatraServlet with JacksonJsonSupport with RequestHeaderAuthentication {
+
+  case object DateTimeSerializer extends CustomSerializer[DateTime](format => ( {
+    case _ => throw new NotImplementedError("DateTime deserialization")
+  }, {
+    case d: DateTime => JString(d.toString(DateTimePropertyFormat))
+  }))
+
+  protected implicit val jsonFormats: Formats = DefaultFormats + DateTimeSerializer
+
+  lazy val csvDataExporter = new CsvDataExporter( eventBus)
+  lazy val assetReportCsvExporter =  new AssetReportCsvExporter(roadLinkService, eventBus, userProvider)
+  val importLogDao: ExportReportDAO = new ExportReportDAO
+
+  before() {
+    contentType = formats("json")
+    try {
+      authenticateForApi(request)(userProvider)
+    } catch {
+      case ise: IllegalStateException => halt(Unauthorized("Authentication error: " + ise.getMessage))
+    }
+    response.setHeader("Digiroad2-Server-Originated-Response", "true")
+  }
+
+  get("/downloadCsv/:id") {
+    val id = params("id")
+
+    if (id.trim.isEmpty) {
+      ""
+    } else {
+      val data = csvDataExporter.getExportById(id.toLong)
+
+      data match {
+        case  Some(info) => info
+        case _ =>
+      }
+    }
+
+  }
+
+  get("/log") {
+    csvDataExporter.getByUser(userProvider.getCurrentUser().username)
+  }
+
+  get("/logs/:id") {
+    val id = params("id").toLong
+    csvDataExporter.getById(id).getOrElse("Logia ei löytynyt.")
+  }
+
+  get("/logs/:ids") {
+    val ids = params("ids").split(',').map(_.toLong).toSet
+    csvDataExporter.getByIds(ids)
+  }
```

Comment:
This here won't work. They are using the same entry point so it will always match with the same one. Since this is using ScalatraBase it will order the entry points from the end of the document to the start therefore it will always match with 
``` Scala 
get("/logs/:ids") {

Reply:
fixed.